packngo.go

```diff
@@ -330,6 +335,40 @@ func NewClientWithAuth(consumerToken string, apiKey string, httpClient *retryabl
 	return client
 }
 
+func PacketRetryPolicy(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	// do not retry on context.Canceled or context.DeadlineExceeded
+	if ctx.Err() != nil {
+		return false, ctx.Err()
+	}
+
+	if err != nil {
+		if v, ok := err.(*url.Error); ok {
+			// Don't retry if the error was due to too many redirects.
+			if redirectsErrorRe.MatchString(v.Error()) {
+				return false, nil
+			}
+
+			// Don't retry if the error was due to TLS cert verification failure.
+			if _, ok := v.Err.(x509.UnknownAuthorityError); ok {
+				return false, nil
+			}
+		}
+
+		// The error is likely recoverable so retry.
+		return true, nil
+	}
+
+	// Check the response code. We retry on 500-range responses to allow
+	// the server time to recover, as 500's are typically not permanent
+	// errors and may relate to outages on the server side. This will catch
+	// invalid response codes as well, like 0 and 999.
+	//if resp.StatusCode == 0 || (resp.StatusCode >= 500 && resp.StatusCode != 501) {
```

Comment:
What happens when we get real 503 like reported here https://github.com/terraform-providers/terraform-provider-packet/issues/149

Reply:
The reported network error in that issue is not HTTP 503, but a TCP error "opeeration timed out", I think that's when the client code is trying to open TCP socket to api.packet.net:443 and it can't for a long time.

The retryable http client will retry in such case. And I think that's alright, as that can happen when your train's in a tunnel, or when the infrastructure service api.packet.net is suddenly down. Both situations are most likely not permanent, so it makes sense to retry.

But I agree it's hard to say what is recoverable and what not. It shouldn't be a guessing game, I wuold like to get some feedback on this topic in here:
https://github.com/terraform-providers/terraform-provider-packet/pull/194